.github/workflows/phpstan.yml

```diff
@@ -0,0 +1,47 @@
+name: 'PHPStan Tests'
+on:
+  pull_request:
+    paths:
+      - 'src/**.php'
+      - '*.php'
+jobs:
+  phpstan:
+    strategy:
+      matrix:
+        phpVersion: [
+            "7.4",
+            "8.0",
+            "8.1",
+            "8.2",
+            "8.3",
+        ]
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1000
+          submodules: recursive
+      # ------------------------------------------------------------------------------
+      # Set up PHP and run phpstan
+      # ------------------------------------------------------------------------------
+      - name: Configure PHP environment to run phpstan
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.phpVersion }}
+      # ------------------------------------------------------------------------------
+      # Override composer php version
```

Comment:
Since we don't set a php version in phpstan, it looks for the composer file.
Setting this overrides the one in composer, so it matches the matrix version we are testing.